src/main/java/com/microsoft/sqlserver/jdbc/dtv.java

```diff
@@ -2063,7 +2063,8 @@ else if (type.isBinary()) {
             // then do the conversion now so that the decision to use a "short" or "long"
             // SSType (i.e. VARCHAR vs. TEXT/VARCHAR(max)) is based on the exact length of
             // the MBCS value (in bytes).
-            else if (null != collation && (JDBCType.CHAR == type || JDBCType.VARCHAR == type
+            // If useBulkCopyForBatchInsert is true, conversion to byte array is not done due to performance
+            else if ((con.getUseBulkCopyForBatchInsert() == false) && null != collation && (JDBCType.CHAR == type || JDBCType.VARCHAR == type
```

Comment:
Instead of else if ((con.getUseBulkCopyForBatchInsert() == false)), we can use:

```
else if ((!con.getUseBulkCopyForBatchInsert()) 
        && collation != null 
        && (JDBCType.CHAR == type || JDBCType.VARCHAR == type))

```